### PR TITLE
Adds hotgraphic pin aria labeling

### PR DESCRIPTION
**Pull request by amygroshek**
Adds aria-haspopup attribute and abstracts a portion of the
aria-label to the schema so the pin description can be
changed in course.json.

See #148  and #149. This only addresses pin labeling, under the assumption that focus and labeling for the dialog will be addressed in the notify core plugin.

### CODE REVIEW

File: properties.schema [hunk 1]
```diff
@@ -19,6 +19,14 @@
       "inputType": "Text",
       "validators": [],
       "translatable": true
+    },
+      "ariaPinLabel": {
+      "type": "string",
+      "required": true,
+      "default": "Hotgraphic pin number ",
```
Comment (moloko) [inline properties.schema]: Is a learner going to know what a 'hotgraphic' is? "Clickable pin" or "Selectable pin" might be better and then gives the learner a strong indication of what it does.

File: templates/hotgraphic.hbs [hunk 2]
```diff
@@ -54,7 +54,7 @@
       {{#if _useGraphicsAsPins}}
         <div class="hotgraphic-narrative">
           {{#each _items}}
-          <button class="base hotgraphic-graphic-pin component-item-text-color {{#if _isVisited}}visited{{/if}} hotgraphic-graphic-pin-{{@index}} {{#if _graphic._classes}}{{_graphic._classes}}{{/if}}" data-id="item-{{@index}}" style="top:{{{_top}}}%; left:{{{_left}}}%;" role="button" aria-label="Item {{numbers @index}}. {{title}}.{{#if _isVisited}} {{../../_globals._accessibility._ariaLabels.visited}}.{{/if}}">
+          <button class="base hotgraphic-graphic-pin component-item-text-color {{#if _isVisited}}visited{{/if}} hotgraphic-graphic-pin-{{@index}} {{#if _graphic._classes}}{{_graphic._classes}}{{/if}}" data-id="item-{{@index}}" style="top:{{{_top}}}%; left:{{{_left}}}%;" role="button" aria-haspopup="true" aria-label="{{../_globals._components._hotgraphic.ariaPinLabel}}  {{numbers @index}}. {{title}}.{{#if _isVisited}} {{../../_globals._accessibility._ariaLabels.visited}}.{{/if}}">
```
Comment (moloko) [inline templates/hotgraphic.hbs]: according to the [docs](https://www.w3.org/TR/wai-aria/#aria-haspopup) a value of `"true"` "indicates the popup is a menu" - this should be set to `"dialog"` and we'll need to make a change elsewhere to ensure that the Notify popup has a (matching) role of `"dialog"` too.

**Comment by amygroshek** [comment]
@moloko I've pushed a fix for this and also created an issue. I am reluctant to rename the branch in accordance with your "issues/#" convention since your PR is already recorded here. I'm happy to do so if you feel it's necessary, however.

**Comment by moloko** [comment]
@amygroshek no need to rename the branch

File: templates/hotgraphic.hbs [hunk 3]
```diff
@@ -17,7 +17,8 @@
...
         {{/if}}
         {{#each _items}}
-          <button class="base hotgraphic-graphic-pin component-item-text-color {{#if _isVisited}}visited{{/if}}" data-index="{{@index}}" style="top:{{{_top}}}%; left:{{{_left}}}%;" role="button" aria-label="Item {{numbers @index}}. {{title}}.{{#if _isVisited}} {{../../_globals._accessibility._ariaLabels.visited}}.{{/if}}">
+          <button class="base hotgraphic-graphic-pin component-item-text-color item-{{@index}} {{#if _isVisited}}visited{{/if}}" data-id="item-{{@index}}" style="top:{{{_top}}}%; left:{{{_left}}}%;" role="button" aria-haspopup="dialog" aria-label="{{../_globals._components._hotgraphic.ariaPinLabel}} {{numbers @index}}. {{title}}.{{#if _isVisited}} {{../_globals._accessibility._ariaLabels.visited}}.{{/if}}">
+          <button class="base hotgraphic-graphic-pin component-item-text-color {{#if _isVisited}}visited{{/if}}" data-index="{{@index}}" style="top:{{{_top}}}%; left:{{{_left}}}%;" role="button" aria-label="{{../_globals._components._hotgraphic.ariaPinLabel}} {{numbers @index}}. {{title}}.{{#if _isVisited}} {{../_globals._accessibility._ariaLabels.visited}}.{{/if}}">
```
Comment (oliverfoster) [inline templates/hotgraphic.hbs]: oop, double button here

Comment (moloko) [reply templates/hotgraphic.hbs]: dammit my mistake when trying to resolve the conflict, I might just redo the PR as I can't edit these files now

**Comment by moloko** [comment]
@amygroshek I've screwed this up with my ham-fisted attempt to resolve a Git conflict so have re-done in #154 rather than ask you to do anything further on this!

**Comment by amygroshek** [comment]
@moloko All good. :)

**Pull request closed without merging**